src/app/components/TableCoins.tsx

```diff
@@ -9,60 +10,84 @@ import CoinsMarketStats from "./CoinsMarketStats";
 
 export default function TableCoins() {
   const dispatch: AppDispatch = useDispatch();
-  const { coins, hasError, currentPage } = useAppSelector(
+  const { coins, loading, hasError, currentPage } = useAppSelector(
     (state) => state.coinsTable
   );
 
   const currencyCode = useAppSelector(
     (state) => state.currency.currentCurrency.code
   );
 
-  useEffect(() => {
-    dispatch(getCoinMarketData({ currency: currencyCode, page: currentPage }));
-  }, [dispatch, currencyCode, currentPage]);
+  /* useEffect(() => {
+    if (coins.length === 0 && loading !== "pending") {
+      // Checks if there are no coins and nothing is currently loading
+      dispatch(
+        getCoinMarketData({ currency: currencyCode, page: currentPage })
+      );
+    }
+  }, [dispatch, currencyCode]); */
+  //When I use useEffect it fetches 1st page two times, also currency dropdown selector doesn't work.
+  //When i try to change to different currency from the dropdown currency menu it doesn't fetch data for the selected currency.
+
+  const fetchMoreData = () => {
+    if (loading !== "pending") {
```

Comment:
Why do we have loading !== "pending". Does pending and loading don't mean same thing in this context?

I usually have this check `if (loading) {}`